### PR TITLE
Stage/unstage files with double-click

### PR DESCRIPTION
**Pull request by Philipp91**
This time without introducing a delay for normal clicks.

### CODE REVIEW

**Comment by JetpackDuba** [comment]
Hey thanks for a new PR!

The click works better but there is a bug (I'll add a comment in the code) and a few UI regressions:
- The hover does no longer work.
- The ripple has also disappeared.

**Review by JetpackDuba — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/jetpackduba/gitnuro/ui/UncommitedChanges.kt [hunk 1]
```diff
@@ -618,7 +618,8 @@ private fun FileEntry(
 
     Box(
         modifier = Modifier
-            .handMouseClickable { onClick() }
+            .handOnHover()
+            .pointerInput(Unit) { detectTapGestures(onPress = { onClick() }, onDoubleTap = { onButtonClick() }) }
```
[the remark this answers is not in the document]
Comment (Philipp91) [reply src/main/kotlin/com/jetpackduba/gitnuro/ui/UncommitedChanges.kt]: So #80 won't be enough to fix this? Basically it's an outer key, but that should keep things apart properly, as it won't re-compose the same Composable instance with different StatusEntries.

(It'd be strange to hand down the `statusEntry` into the new `Modifier.clickableWithDoubleClick()` helper.)

**Comment by Philipp91** [comment]
> The hover does no longer work.

I assume you mean the "Stage" button that appears when hovering the entry. I can't reproduce this issue.

> The ripple has also disappeared.

Fixed.

**Comment by JetpackDuba** [comment]
> > The hover does no longer work.
> 
> I assume you mean the "Stage" button that appears when hovering the entry. I can't reproduce this issue.

No, I mean the hover effect
imagen

It's more visible when using the light theme

imagen

**Comment by Philipp91** [comment]
Huh, where does that hover effect even come from? And where is its color defined?

**Comment by JetpackDuba** [comment]
Now that you mention it... If my memory doesn't fail, it's something done by compose desktop itself, yikes.

**Comment by Philipp91** [comment]
I could reverse-engineer the color, put it as a new entry into the Gitnuro theme and use another `backgroundIf`.

Actually I think it's not a plain color, but actually white or black (depending on dark/light theme) or a medium gray with some opacity. That's why it also works on top of the "selected" background.

**Comment by JetpackDuba** [comment]
Nah, don't worry, we can just pick a new color instead. I'll look into it tomorrow.

**Comment by Philipp91** [comment]
I found that the hover color isn't the only thing that's destroyed by this approach. I've implemented another approch now, which tries to interfere with the standard `clickable()` functionalities as little as possible.

**Comment by JetpackDuba** [comment]
Hey!

Sorry for the late response. The last days I've been testing the changes to see if the UX is what I was expecting. The last change was great, as the delay is no longer present and displaying the diff before the second click doesn't seem like a big deal, as the double click makes most sense when the diff is already open (click to see the diff -> double click to (un)stage the file without having to move the cursor to the button).

**Pull request merged**